src/main/java/org/gwtproject/activity/widget/shared/WidgetActivity.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2010 Google Inc.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.gwtproject.activity.widget.shared;
+
+import org.gwtproject.activity.shared.Activity;
+import org.gwtproject.activity.shared.ActivityDisplay;
+import org.gwtproject.event.shared.EventBus;
+import org.gwtproject.user.client.ui.AcceptsOneWidget;
+import org.gwtproject.user.client.ui.IsWidget;
+
+/**
+ * Drop-in replacement for {@link com.google.gwt.activity.shared.Activity}.
+ */
+public interface WidgetActivity extends Activity<IsWidget> {
+
+  default void start(ActivityDisplay<IsWidget> panel, EventBus eventBus) {
+    start((AcceptsOneWidget) panel::show, eventBus);
```

Comment:
Like I said, I would trade simplicity over microoptimization any time, but like you pointed out, this may actually break activities if they compare panels (didn't occur to me they may), so I agree this is safer. Will do.

Reply:
Looks like activities never had an opportunity to compare panels they got, since `ActivityManager` wraps every display/panel with new `ProtectedDisplay` every time `Activity.start` is invoked (see `ActivityManager.tryStart`).

So, (double) wrapping seems inevitable.

Reply:
And on the other hand, that simplifies the code a lot, as `setDisplay(AcceptsOnWidget panel)` can then just call `setDisplay(panel == null ? null : panel::setWidget)` (i.e. you can remove the `widgetDisplay` and `wrapperDisplay`)